flang-new: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'

## Description

The `-fPIC` option is automatically added to the compiler flags by `fpm`, but `flang-new` does not support this option on the `x86_64-pc-windows-msvc` target.

```
Run fpm test --compiler flang-new --profile debug --verbose
 <INFO> BUILD_NAME: build\flang-new
 <INFO> COMPILER:  flang-new
 <INFO> C COMPILER:  clang
 <INFO> CXX COMPILER: clang++
 <INFO> COMPILER OPTIONS:   -fPIC
 <INFO> C COMPILER OPTIONS:  
 <INFO> CXX COMPILER OPTIONS: 
 <INFO> LINKER OPTIONS:  
 <INFO> INCLUDE DIRECTORIES:  []
 + mkdir build\flang-new_1C9F1A78448FC13A
 + mkdir build\flang-new_68BEE2AA1F5C26E0
[  0%]               module_hello.f90
 + mkdir build\flang-new_1C9F1A78448FC13A\test\
 + flang-new -c .\.\src\module_hello.f90   -fPIC -module-dir build\flang-new_1C9F1A78448FC13A -Ibuild\flang-new_1C9F1A78448FC13A -o build\flang-new_1C9F1A78448FC13A\test\src_module_hello.f90.o
flang-new: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
flang-new: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
[ 25%]               module_hello.f90  done.

<ERROR> Compilation failed for object " src_module_hello.f90.o "
<ERROR> stopping due to failed compilation
STOP 1
flang-new: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
flang-new: error: unsupported option '-fPIC' for target 'x86_64-pc-windows-msvc'
Error: Process completed with exit code 1.

```

### Expected Behaviour

fpm should avoid adding -fPIC when the target platform does not support or it.

without `-fpic`:
```
Run fpm test --compiler flang-new --flag "-g" --verbose
 <INFO> BUILD_NAME: build\flang-new
 <INFO> COMPILER:  flang-new
 <INFO> C COMPILER:  clang
 <INFO> CXX COMPILER: clang++
 <INFO> COMPILER OPTIONS:   -g
 <INFO> C COMPILER OPTIONS:  
 <INFO> CXX COMPILER OPTIONS: 
 <INFO> LINKER OPTIONS:  
 <INFO> INCLUDE DIRECTORIES:  []
 + mkdir build\flang-new_D0EC3A23ED75F785
 + mkdir build\flang-new_45AE5EB34B9F9B4B
[  0%]               module_hello.f90
 + mkdir build\flang-new_D0EC3A23ED75F785\test\
 + flang-new -c .\.\src\module_hello.f90   -g -module-dir build\flang-new_D0EC3A23ED75F785 -Ibuild\flang-new_D0EC3A23ED75F785 -o build\flang-new_D0EC3A23ED75F785\test\src_module_hello.f90.o
[ 25%]               module_hello.f90  done.
[ 25%]                      libtest.a
 + ar -rs build\flang-new_D0EC3A23ED75F785\test\libtest.a @build\flang-new_D0EC3A23ED75F785\test\libtest.a.resp
ar: creating build\flang-new_D0EC3A23ED75F785\test\libtest.a
[ 50%]                      libtest.a  done.
[ 50%]                      check.f90
 + flang-new -c test\check.f90   -g -module-dir build\flang-new_D0EC3A23ED75F785 -Ibuild\flang-new_D0EC3A23ED75F785 -o build\flang-new_D0EC3A23ED75F785\test\test_check.f90.o
[ 75%]                      check.f90  done.
[ 75%]                      check.exe
 + mkdir build\flang-new_45AE5EB34B9F9B4B\test\
 + flang-new    -g  build\flang-new_D0EC3A23ED75F785\test\test_check.f90.o  build\flang-new_D0EC3A23ED75F785\test\libtest.a -o build\flang-new_45AE5EB34B9F9B4B\test\check.exe
[100%]                      check.exe  done.
[100%] Project compiled successfully.
 + build\flang-new_45AE5EB34B9F9B4B\test\check.exe 
Hello World!
```

### Version of fpm

0.12.0, alpha

### Platform and Architecture

Windows

### Additional Information

_No response_

--- a/src/fpm_compiler.F90
+++ b/src/fpm_compiler.F90
@@ -288,10 +288,17 @@ function get_default_flags(self, release) result(flags)
     ! Append position-independent code (PIC) flag, that is necessary 
     ! building shared libraries
     select case (self%id)
-    case (id_gcc, id_f95, id_caf, id_flang, id_flang_new, id_f18, id_lfortran, &
+    case (id_gcc, id_f95, id_caf, id_flang, id_f18, id_lfortran, &
           id_intel_classic_nix, id_intel_classic_mac, id_intel_llvm_nix, &
           id_pgi, id_nvhpc, id_nag, id_cray, id_ibmxl)
         pic_flag = " -fPIC"
+    case (id_flang_new)
+        ! flang-new doesn't support -fPIC on Windows MSVC target
+        if (get_os_type() == OS_WINDOWS) then
+            pic_flag = ""
+        else
+            pic_flag = " -fPIC"
+        end if
     case (id_intel_classic_windows, id_intel_llvm_windows)
         pic_flag = ""  ! Windows does not use -fPIC
     case default

